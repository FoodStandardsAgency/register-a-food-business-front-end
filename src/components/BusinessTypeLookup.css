div.errorStyling {
  border-left: 4px solid #b10e1e;
  padding-left: 10px;
}

.errorStyling .autocomplete__input {
  border: 3px solid #b10e1e !important;
}
