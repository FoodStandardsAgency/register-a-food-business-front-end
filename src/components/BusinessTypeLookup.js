import dynamic from "next/dynamic";
import "accessible-autocomplete/dist/accessible-autocomplete.min.css";
import "./BusinessTypeLookup.css";

import {
  findMatches,
  inputValueFunction,
  suggestionFunction
} from "./BusinessTypeLookupFunctions";

import { ErrorText, HintText } from "govuk-react";

// dynamic import used because Autocomplete component from AlphaGov uses the document object on import.
// Therefore it must be imported on the client side not on the server side.
// TODO JMB: contribute back to the AlphaGov repository to avoid or delay reliance on the document object.
const Autocomplete = dynamic(import("accessible-autocomplete/react"));

const templates = {
  inputValue: inputValueFunction,
  suggestion: suggestionFunction
};

const lineHeight = {
  lineHeight: "1.6em",
  padding: 0
};

const BusinessTypeLookup = props => (
  <div className={props.validatorErrors.business_type ? "errorStyling" : null}>
    <HintText style={lineHeight}>
      For example cafe, hospital canteen or commercial bakery
    </HintText>
    {props.validatorErrors.business_type ? (
      <ErrorText style={lineHeight}>
        {props.validatorErrors.business_type}
      </ErrorText>
    ) : null}
    <Autocomplete
      source={findMatches}
      templates={templates}
      autoselect={true}
      confirmOnBlur={false}
      name="business_type"
      defaultValue={props.cumulativeAnswers.business_type}
    />
  </div>
);

export default BusinessTypeLookup;
