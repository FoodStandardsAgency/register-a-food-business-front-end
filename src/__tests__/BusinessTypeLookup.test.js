import BusinessTypeLookup from "../components/BusinessTypeLookup";
import { shallow } from "enzyme";

describe("<BusinessTypeLookup />", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(
      <BusinessTypeLookup validatorErrors cumulativeFullAnswers />
    );
    expect(wrapper.length).toBe(1);
  });

  describe("given validatorErrors.business_type", () => {
    it("renders the container div with different styling", () => {
      const wrapperNoErrors = shallow(
        <BusinessTypeLookup validatorErrors={{}} cumulativeFullAnswers />
      );
      const appliedClassNameNoErrors = wrapperNoErrors
        .find("#autocompleteContainer")
        .prop("className");
      const wrapperErrors = shallow(
        <BusinessTypeLookup
          validatorErrors={{ business_type: "An error" }}
          cumulativeFullAnswers
        />
      );
      const appliedClassNameErrors = wrapperErrors
        .find("#autocompleteContainer")
        .prop("className");

      expect(appliedClassNameNoErrors).not.toEqual(appliedClassNameErrors);
    });
  });
});
