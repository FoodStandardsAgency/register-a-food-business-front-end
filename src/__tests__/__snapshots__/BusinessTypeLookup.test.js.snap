// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BusinessTypeLookup /> matches the previous snapshot 1`] = `
.emotion-0 {
  display: block;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  text-transform: none;
  font-size: 16px;
  line-height: 1.25;
  color: #6f777b;
  margin-bottom: 0;
}

@media only screen and (min-width:641px) {
  .emotion-0 {
    font-size: 19px;
    line-height: 1.3;
  }
}

@media only screen and (min-width:641px) {
  .emotion-0 {
    margin-bottom: 0;
  }
}

<div
  className={null}
>
  <span
    className="emotion-0"
    style={
      Object {
        "lineHeight": "1.6em",
        "padding": 0,
      }
    }
  >
    For example cafe, hospital canteen or commercial bakery
  </span>
  <div
    aria-expanded="false"
    className="autocomplete__wrapper"
    onKeyDown={[Function]}
    role="combobox"
  >
    <div
      aria-atomic="true"
      aria-live="polite"
      role="status"
      style={
        Object {
          "border": "0",
          "clip": "rect(0 0 0 0)",
          "height": "1px",
          "marginBottom": "-1px",
          "marginRight": "-1px",
          "overflow": "hidden",
          "padding": "0",
          "position": "absolute",
          "whiteSpace": "nowrap",
          "width": "1px",
        }
      }
    >
      No search results.
      <span>
        ,,
      </span>
    </div>
    
    <input
      aria-activedescendant={false}
      aria-owns="undefined__listbox"
      autoComplete="off"
      className="autocomplete__input autocomplete__input--default"
      id={undefined}
      name="business_type"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      placeholder=""
      required={false}
      role="textbox"
      type="text"
      value=""
    />
    <ul
      className="autocomplete__menu autocomplete__menu--inline autocomplete__menu--hidden"
      id="undefined__listbox"
      role="listbox"
    />
  </div>
</div>
`;
